Merge back rebased 0.14.0

phil-davis opened the pull request:
These are the changes that actually need to be merged back from `release-0.14.0` branch.

To get this I did:

```
git checkout release-0.14.0
git checkout -b rebased-0.14.0
git rebase master
First, rewinding head to replay your work on top of it...
Applying: add changelog items and bump version
Applying: Hotfix #447
Applying: Disable phan on 0.14.0 due to PR 449
Applying: Update CHANGELOG.md
```
It found just 4 commits that were needed - all the other commits match with commits already in master.

Then:
```
git revert ff9f73ca6975c6f411865b4c23e491c80e400c19
[rebased-0.14.0 960f4ef] Revert "Hotfix #447"
 2 files changed, 1 insertion(+), 42 deletions(-)
 delete mode 100644 appinfo/register_command.php
git revert 5bdce7e02299f985201f0742340b0a74becfc1d3
[rebased-0.14.0 325ec0c] Revert "Disable phan on 0.14.0 due to PR 449"
 2 files changed, 673 insertions(+), 1 deletion(-)
```
That reverts the stuff that we actually do not want in master.

Now we have just the changelog and version bump, which I think is all that needs to be applied to master.

micbar commented:
@phil-davis Did you check where the tag (v0.14.0) is attached to?
We need to be sure that it points to the right commitID and that this commitID is present here in this branch.

phil-davis commented:
```
git checkout release-0.14.0 
Switched to branch 'release-0.14.0'
Your branch is up to date with 'origin/release-0.14.0'.
git log -1
commit a9e766d0108da37b233ab8fd86d9e9d8eef2a316 (HEAD -> release-0.14.0, tag: v0.14.0, origin/release-0.14.0)
Author: Michael Barz <mbarz@owncloud.com>
Date:   Fri Nov 8 12:10:08 2019 +0100

    Update CHANGELOG.md
```
The tag is correctly at the same commit as the end of `release-0.14.0`

```
git checkout rebased-0.14.0 
Switched to branch 'rebased-0.14.0'
Your branch is up to date with 'origin/rebased-0.14.0'.
git log -1
commit 325ec0ce0e532de57a283e827f3e64d5cbde43ad (HEAD -> rebased-0.14.0, origin/rebased-0.14.0)
Author: Phil Davis <phil@jankaritech.com>
Date:   Fri Nov 8 18:16:29 2019 +0545

    Revert "Disable phan on 0.14.0 due to PR 449"
    
    This reverts commit 5bdce7e02299f985201f0742340b0a74becfc1d3.
```
Branch `rebased-0.14.0` is different, which is expected.
